### PR TITLE
Documentation fixups

### LINKED ISSUE #1755: some documentation fixups
## Description
The boost_policy has three custom settings, only two are documented, which is kinda weird. Probably "boost::math::policies::promote_double<false>" should get a line in the docstring too.

Also see the change set in #1756 for typos and copypasta.

## Example
N/A.

## Expected Output
N/A.

#### Current Version:
v3.1.0

### PR DESCRIPTION
**Pull request by peterwicksstringfield**
## Summary

Copypasta and typos. Document the third option in the boost_policy.

Shout out to this tool that I used to find the typos https://github.com/louis-langholtz/pyspellcode.

Closes #1755.

## Tests

None.

## Side Effects

Nope.

## Release Notes
Miscellaneous doxygen documentation cleanup.

I updated RELEASE_NOTES.md too. Probably we don't need a release notes entry for typos, but at release time its easier to delete an entry than reconstruct one. If we don't want it I can delete it from this PR without restarting the CI process.

## Checklist

- [X] Math issue #1755
- [X] Copyright holder: Peter Wicks Stringfield

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [X] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [X] the code is written in idiomatic C++ and changes are documented in the doxygen

- [X] the new changes are tested

### CODE REVIEW

**Comment by mcol** [comment]
Thanks, fantastic job! I've been collecting typos too, here are few I found:
- stan/math/opencl/buffer_types.hpp:  only it's original type -> only its original type
- stan/math/opencl/kernels/check_symmetric.hpp: wheter -> whether
- stan/math/opencl/kernels/triangular_transpose.hpp: to it's upper/lower -> to its upper/lower
- stan/math/opencl/rev/triangular_transpose.hpp: to it's upper/lower -> to its upper/lower
- stan/math/opencl/triangular_transpose.hpp: to it's upper/lower -> to its upper/lower
- stan/math/prim/fun/grad_reg_lower_inc_gamma.hpp:  w.r.t it's first parameter ->  w.r.t its first parameter
- stan/math/prim/meta/require_generics.hpp: and it's inner type -> and its inner type
- stan/math/rev/fun/LDLT_factor.hpp: that it's lifespan -> that its lifespan
- `Faiulre`, `thorwing` and `thorws` in tests
Let me know if you prefer that I push to your branch rather than fixing them yourself.

File: stan/math/fwd/core/operator_logical_and.hpp [hunk 1]
```diff
@@ -13,7 +13,7 @@ namespace math {
  * @tparam value and tangent type for variables
  * @param[in] x first argument
  * @param[in] y second argument
- * @return disjuntion of the argument's values
+ * @return conjunction of the argument's values
```
Comment (mcol) [inline stan/math/fwd/core/operator_logical_and.hpp]: This should be `arguments' values`.

File: stan/math/fwd/core/operator_logical_or.hpp [hunk 2]
```diff
@@ -13,7 +13,7 @@ namespace math {
  * @tparam value and tangent type for variables
  * @param[in] x first argument
  * @param[in] y second argument
- * @return disjuntion of the argument's values
+ * @return disjunction of the argument's values
```
Comment (mcol) [inline stan/math/fwd/core/operator_logical_or.hpp]: This should be `arguments' values`.

File: stan/math/prim/fun/inv_sqrt.hpp [hunk 3]
```diff
@@ -46,7 +46,7 @@ inline auto inv_sqrt(const T& x) {
  *
  * @tparam Derived derived type of x
  * @param x Matrix or matrix expression
- * @return Arc cosine of each variable in the container, in radians.
+ * @return 1 / sqrt of each value in x.
```
Comment (mcol) [inline stan/math/prim/fun/inv_sqrt.hpp]: [optional] I would say "The inverse square root ..." (same for the other instance in this file).

File: stan/math/rev/core/operator_logical_and.hpp [hunk 4]
```diff
@@ -13,7 +13,7 @@ namespace math {
  *
  * @param[in] x first argument
  * @param[in] y second argument
- * @return disjuntion of the argument's values
+ * @return conjunction of the argument's values
```
Comment (mcol) [inline stan/math/rev/core/operator_logical_and.hpp]: Should be `arguments' values`.

File: stan/math/rev/core/operator_logical_or.hpp [hunk 5]
```diff
@@ -13,7 +13,7 @@ namespace math {
  *
  * @param[in] x first argument
  * @param[in] y second argument
- * @return disjuntion of the argument's values
+ * @return disjunction of the argument's values
```
Comment (mcol) [inline stan/math/rev/core/operator_logical_or.hpp]: Should be `arguments' values`.

**Comment by peterwicksstringfield** [comment]
I plan to fix all the extra typos you found tomorrow. Thanks for finding all that, especially the ones that are literally on the same line as ones I fixed. (I knew that was going to happen, no matter how many times I skimmed the diff.)

(That said, you are welcome to push to this branch if you want, but it is not necessary for you to do so.)

**Comment by mcol** [comment]
There's no hurry. By the way, I've also spotted a few "paramater"...

File: stan/math/fwd/core/operator_logical_and.hpp (hunk 1 above)
Comment (peterwicksstringfield) [reply stan/math/fwd/core/operator_logical_and.hpp]: Done.

File: stan/math/fwd/core/operator_logical_or.hpp (hunk 2 above)
Comment (peterwicksstringfield) [reply stan/math/fwd/core/operator_logical_or.hpp]: Done.

File: stan/math/rev/core/operator_logical_or.hpp (hunk 5 above)
Comment (peterwicksstringfield) [reply stan/math/rev/core/operator_logical_or.hpp]: Done.

File: stan/math/rev/core/operator_logical_and.hpp (hunk 4 above)
Comment (peterwicksstringfield) [reply stan/math/rev/core/operator_logical_and.hpp]: Done.

File: stan/math/prim/fun/inv_sqrt.hpp (hunk 3 above)
Comment (peterwicksstringfield) [reply stan/math/prim/fun/inv_sqrt.hpp]: Done.

**Comment by peterwicksstringfield** [comment]
Somehow "paramater" got into my whitelist. Thanks. I fixed everything else you mentioned as well.

**Comment by mcol** [comment]
Thanks a lot!

**Pull request merged**